drivers/serial/uart_ns16550.c

```diff
@@ -480,13 +497,18 @@ static int uart_ns16550_init(struct device *dev)
  */
 static int uart_ns16550_poll_in(struct device *dev, unsigned char *c)
 {
+	k_spinlock_key_t key = k_spin_lock(&DEV_DATA(dev)->lock);
+
```

Comment:
This could be refactored to:
```
int rv = -1;
k_spinlock_key_t key = k_spin_lock(&DEV_DATA(dev)->lock);

if ((INBYTE(LSR(dev)) & LSR_RXRDY) != 0) {
    /* got a character */
￼   *c = INBYTE(RDR(dev)); 
    rv = 0;
}

k_spin_unlock(&DEV_DATA(dev)->lock, key);

return rv;
````

It's generally easier to pair up lock/unlock operations when they occur in pairs at the same depth.

Not a required change.

Reply:
Awesome, thanks for the code suggestion and dev tips for spinlocks!